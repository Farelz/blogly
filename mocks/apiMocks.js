export const fetch_data_mock = {
    data: {
        allPost: {
            edges: [
                {
                    node: {
                        id: 'UG9zdDox',
                        uuid: '1',
                        title: 'Hello World',
                        body: '<div><p>Hey there im new</p>\n</div>',
                        timestamp: '2018-06-05T01:28:35.802157',
                        postPicUrl: '',
                        views: 16,
                        author: {
                            id: 'VXNlcjoy',
                            fullName: 'Marvin Kome',
                            description: null,
                            gravatarUrl: null
                        },
                        tags: {
                            edges: [
                                {
                                    node: {
                                        id: 'VGFnczox',
                                        name: 'hello'
                                    }
                                }
                            ]
                        },
                        claps: {
                            totalCount: 0
                        },
                        comments: {
                            edges: [
                                {
                                    node: {
                                        id: 'Q29tbWVudDox',
                                        uuid: '1',
                                        body: 'Hello World',
                                        timestamp: '2018-06-05T01:30:59.729280',
                                        author: {
                                            id: 'VXNlcjox',
                                            fullName: 'Fake User',
                                            gravatarUrl: null
                                        },
                                        replies: {
                                            edges: []
                                        }
                                    }
                                },
                                {
                                    node: {
                                        id: 'Q29tbWVudDoy',
                                        uuid: '2',
                                        body: 'Notify Marvin Correctly',
                                        timestamp: '2018-06-05T02:09:39.718090',
                                        author: {
                                            id: 'VXNlcjox',
                                            fullName: 'Fake User',
                                            gravatarUrl: null
                                        },
                                        replies: {
                                            edges: []
                                        }
                                    }
                                },
                                {
                                    node: {
                                        id: 'Q29tbWVudDoz',
                                        uuid: '3',
                                        body: 'Hello Marvin',
                                        timestamp: '2018-06-05T11:58:49.746609',
                                        author: {
                                            id: 'VXNlcjox',
                                            fullName: 'Fake User',
                                            gravatarUrl: null
                                        },
                                        replies: {
                                            edges: []
                                        }
                                    }
                                },
                                {
                                    node: {
                                        id: 'Q29tbWVudDo0',
                                        uuid: '4',
                                        body: 'Hello Marvin',
                                        timestamp: '2018-06-09T01:46:10.115571',
                                        author: {
                                            id: 'VXNlcjox',
                                            fullName: 'Fake User',
                                            gravatarUrl: null
                                        },
                                        replies: {
                                            edges: []
                                        }
                                    }
                                }
                            ]
                        }
                    }
                },
            ],
            pageInfo: {
                endCursor: 'YXJyYXljb25uZWN0aW9uOjQ=',
                hasNextPage: false
            }
        }
    }
};
