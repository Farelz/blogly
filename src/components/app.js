/**
 * ./src/components/app
 */

import React from 'react';
import { Route, Switch } from 'react-router-dom';

import Home from './home';
import Post from './post';

const App = () => {
    return (
        <Switch>
            <Route path="/" component={Home} exact />
            <Route path="/post/:id" component={Post} exact />
        </Switch>
    );
};
export default App;
