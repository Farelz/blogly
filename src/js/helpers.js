/**
 * ./app/js/helpers
 */

import moment from 'moment';

export const truncate = (word, length) => {
    const new_word =
        word
            .split(' ')
            .splice(0, length)
            .join(' ') + ' ...';
    return word.split(' ').length < length ? word : new_word;
};

export const format_date = (server_date) => {
    let date = moment.utc(server_date).format('YYYY-MM-DD HH:mm:ss');
    let stillUtc = moment.utc(date).toDate();

    if (
        new Date(date).getFullYear() == new Date().getFullYear() &&
        moment(stillUtc)
            .local()
            .week() !==
            moment(stillUtc)
                .local()
                .week()
    ) {
        return moment(stillUtc)
            .local()
            .format('MMM DD');
    } else if (
        moment(stillUtc)
            .local()
            .week() ==
        moment(stillUtc)
            .local()
            .week()
    ) {
        return moment(stillUtc)
            .local()
            .fromNow();
    }

    return moment(stillUtc)
        .local()
        .format('MMM DD [\']YY');
};

export const validate_password = (password) => {
    const regExp = new RegExp(
        '^(((?=.*[a-z])(?=.*[A-Z]))|((?=.*[a-z])(?=.*[0-9]))|((?=.*[A-Z])(?=.*[0-9])))(?=.{6,})'
    );
    return regExp.test(password);
};

export const validate_html = (html) => {
    const regExp = /<[a-z][\s\S]*>/i;
    return regExp.test(html);
};

export const strip_filename = (name) => {
    return name.split('/').pop();
};

export const upload_file = (file) => {
    const url =
        process.env.NODE_ENV == 'production'
            ? 'https://reactpress-api.herokuapp.com'
            : 'http://0.0.0.0:5000';
    const formData = new FormData();
    formData.append('file', file);

    const headers = {
        method: 'POST',
        body: formData
    };

    return fetch(url, headers).then((res) => res.json());
};

export const gcd = (a, b) => {
    if (b == 0) {
        return a;
    }
    return gcd(b, a % b);
};

export const count_words_in_html = (string) => {
    string = string.replace(/<(?:.|\n)*?>/gm, '');
    string = string.replace(/(^\s*)|(\s*$)/gi, '');
    string = string.replace(/[ ]{2,}/gi, ' ');
    string = string.replace(/\n /, '\n');
    return string.split(' ').length;
};

export const get_profile_link = (fullName, id) => {
    fullName = fullName.toLowerCase();
    let link_list = fullName.split(' ');
    link_list = link_list.concat(id);

    return '/profile/' + link_list.join('_');
};

export const sort_posts = (posts) => {
    if (posts !== undefined) {
        posts.sort((a, b) => {
            if (a.node.timestamp > b.node.timestamp) {
                return -1;
            }
            if (a.node.timestamp < b.node.timestamp) {
                return 1;
            }
            return 0;
        });
    }

    return posts;
};

export const isLoggedIn = () => {
    const sessionLogin = JSON.parse(localStorage.getItem('med-blog-logged-in'));
    const localLogin = sessionLogin != undefined && sessionLogin == true;

    return localLogin;
};
