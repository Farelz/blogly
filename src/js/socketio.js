import openSocket from 'socket.io-client';

const Socket = () => {
    let socket;
    return {
        connect: (token) => {
            socket = openSocket('http://localhost:5000', {
                transportOptions: {
                    polling: {
                        extraHeaders: {
                            Authorization: 'Bearer ' + token
                        }
                    }
                }
            });
        },
        recieve_notification: (cb) => socket.on('notifications', (msg) => cb(msg)),
        read_all_notifications: () => socket.emit('read notifications'), 
        disconnect: () => socket.disconnect()
    };
};

export default Socket;
