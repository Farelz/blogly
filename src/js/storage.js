export const saveToStore = (store, key) => {
    if (localStorage) {
        store = JSON.stringify(store);
        localStorage.setItem(key, store);
    }

    return true;
};

export const removeFromStore = (key) => {
    if (localStorage) {
        localStorage.removeItem(key);
    }

    return true;
};

export const getFromStore = (key) => {
    let value = null;

    if (window.localStorage) {
        value = JSON.parse(window.localStorage.getItem(key));
    }

    return value;
};
