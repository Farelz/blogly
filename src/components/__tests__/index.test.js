import React from 'react';
import { renderRoot } from './test-utils';
import { ConnectApp } from '../app';
import { fetch_data_mock } from '../../../__mocks__/apiMocks.js';
import 'materialize-css/dist/js/materialize.js';

describe('Integration tests', () => {
    const { container } = renderRoot(<ConnectApp/>);

    beforeEach(() => {
        fetch.resetMocks();
    });

    it('renders homepage', () => {
        fetch.mockResponse(JSON.stringify(fetch_data_mock));
        expect(container).toMatchSnapshot();
    });

    // it('is routes to login on click', () => {

    // });
});