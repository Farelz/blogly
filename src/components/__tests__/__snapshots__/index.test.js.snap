// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration tests renders homepage - without mock 1`] = `
<div>
  <div>
    <div
      class="navbar-wrapper"
    >
      <nav
        class="topbar nav-extended"
      >
        <div
          class="nav-wrapper"
        >
          <a
            class="sidenav-trigger hide-on-med-and-up"
            data-target="mobile-topbar"
            href="#"
          >
            <svg
              fill="currentColor"
              height="1em"
              preserveAspectRatio="xMidYMid meet"
              style="vertical-align: middle;"
              viewBox="0 0 40 40"
              width="1em"
            >
              <g>
                <path
                  d="m5 10h30v3.4h-30v-3.4z m0 11.6v-3.2h30v3.2h-30z m0 8.4v-3.4h30v3.4h-30z"
                />
              </g>
            </svg>
          </a>
          <a
            class="brand-logo center"
            href="/"
          >
            <span
              class="blog-title"
            >
              ReactPress
            </span>
          </a>
          <ul
            class="right hide-on-small-only"
          >
            <li>
              <a
                class="sign-in"
                href="/auth/login"
              >
                Sign in
              </a>
            </li>
            <li>
              <a
                class="sign-up btn"
                href="/auth/signup"
              >
                Get started
              </a>
            </li>
          </ul>
        </div>
        <div
          class="nav-content"
        >
          <div>
            <section>
              <span>
                <a>
                  home
                </a>
              </span>
              <span>
                <a>
                  trust
                </a>
              </span>
              <span>
                <a>
                  culture
                </a>
              </span>
              <span>
                <a>
                  tech
                </a>
              </span>
              <span>
                <a>
                  entreprenuership
                </a>
              </span>
              <span>
                <a>
                  self
                </a>
              </span>
              <span>
                <a>
                  politics
                </a>
              </span>
              <span>
                <a>
                  media
                </a>
              </span>
              <span>
                <a>
                  design
                </a>
              </span>
              <span>
                <a>
                  programming
                </a>
              </span>
              <span>
                <a>
                  art
                </a>
              </span>
              <span>
                <a>
                  science
                </a>
              </span>
              <span>
                <a>
                  popular
                </a>
              </span>
            </section>
          </div>
        </div>
      </nav>
      <ul
        class="sidenav hide-on-med-and-up"
        id="mobile-topbar"
      >
        <li>
          <div
            class="user-view"
          >
            <a>
              <img
                class="circle defImg"
                src="test-file-stub"
              />
            </a>
            <span
              class="email"
            >
              Hello, 
              Guest
            </span>
          </div>
        </li>
        <li>
          <a
            class="hide-on-med-and-up"
            href="/auth/login"
          >
            Sign In
          </a>
        </li>
        <div
          class="divider"
        />
        <li>
          <a
            class="hide-on-med-and-up"
            href="/auth/signup"
          >
            Get Started
          </a>
        </li>
      </ul>
    </div>
    <div
      class="body section container"
    >
      <div
        class="row"
      >
        <div
          class="col s12"
        >
          <div
            class="center"
          >
            <h5>
              Getting posts...
            </h5>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
